OIDC scenarion with authorization using `@ClientHeaderParam` doesn't work after move to reactive

### Describe the bug

I'm trying to migrate testing scenarios we have for quarkus-oidc-client-filter / quarkus-oidc-token-propagation / quarkus-rest-client into their reactive variants.

Changes for the move are here: https://github.com/rsvoboda/quarkus-test-suite/commit/818b36371a7f511062377b869fe34407221537f7

There is failure in one scenario called "LookupAuthorization" with authorization using `@ClientHeaderParam` where the Bearer token gets constructed using direct communication with oidc server.
https://github.com/quarkus-qe/quarkus-test-suite/blob/main/security/keycloak-oidc-client-extended/src/main/java/io/quarkus/ts/security/keycloak/oidcclient/extended/restclient/ping/clients/LookupAuthorizationPongClient.java#L25

I can't say for sure if the trouble is in resteasy-reactive, rest-client-reactive or in oidc extensions.
Thus CCing @sberyozkin, @Sgitario (who created that test coverage ;) ) and @geoand 

### Expected behavior

All tests to pass

### Actual behavior

Test failure
```
[ERROR] Failures:
[ERROR]   OidcRestClientIT.testLookupAuthorization:49->assertPingEndpoints:57->assertPingPong:109 1 expectation failed.
Expected status code <200> but was <500>.

[INFO]
[ERROR] Tests run: 9, Failures: 1, Errors: 0, Skipped: 0
```

### How to Reproduce?

 - clone https://github.com/rsvoboda/quarkus-test-suite
 - checkout `oidc-client-reactive-filter` branch
 - run `mvn clean verify -pl security/keycloak-oidc-client-extended`

### Output of `uname -a` or `ver`

macOS Monterey

### Output of `java -version`

Java 17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

Quarkus main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/MicroProfileRestClientEnricher.java
+++ b/extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/MicroProfileRestClientEnricher.java
@@ -445,7 +445,7 @@ class MicroProfileRestClientEnricher implements JaxrsClientReactiveEnricher {
                 for (MethodInfo method : methods) {
                     if (Modifier.isAbstract(method.flags())) {
                         MethodCreator methodCreator = classCreator.getMethodCreator(MethodDescriptor.of(method));
-                        methodCreator.returnValue(methodCreator.loadNull());
+                        methodCreator.throwException(IllegalStateException.class, "This should never be called");
                     }
                 }
             }

--- a/extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/subresource/SubResourceTest.java
+++ b/extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/subresource/SubResourceTest.java
@@ -5,6 +5,7 @@ import static org.assertj.core.api.Assertions.assertThat;
 import java.net.URI;
 
 import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
 import javax.ws.rs.GET;
 import javax.ws.rs.POST;
 import javax.ws.rs.Path;
@@ -106,6 +107,12 @@ public class SubResourceTest {
         @ClientHeaderParam(name = "fromRootMethod", value = "{fillingMethod}")
         @ClientHeaderParam(name = "overridable", value = "RootClient#sub")
         SubClient sub(@PathParam("rootParam") String rootParam, @PathParam("methodParam") String methodParam);
+
+        // we only add this to make sure that the inclusion of @ClientHeaderParam does not break
+        // when the rest client interface contains a method returning a primitive type
+        @Path("/dummy")
+        @DELETE
+        boolean dummy();
 
         default String fillingMethod() {
             return "RootClientComputed";

